buildpacks/python/python/function.go

```diff
@@ -43,10 +71,10 @@ func NewFunction(opts ...FunctionOpt) Function {
 		},
 	)
 
-	return f
+	return &f
```

Comment:
Note that initializers named `New...` should return pointers, so I updated this (and all the contributor initializers)

Reply:
What's the rationale for this?

Reply:
I believe it's for congruency with the `new` keyword. I can't remember exactly where I read it (maybe somewhere in Effective Go), but I heard that most constructors should return pointers, and non-pointer constructors should start with `Make...` (this might help: https://stackoverflow.com/a/18125682/2456938)

Reply:
In our case, we actually want pointers anyway (to pass into the layers array), so it works nicely

Reply:
I think we need to consider here if we really need it to be a reference instead of a value for our usage; else, we the function is essentially a `make` in its old state.

Reply:
That's a valid point. And I think in this case, we could get away with either. I noticed that some of our contributors had pointer receivers and others didn't, so I'm aiming mostly for consistency. The contributors are stateful, so I leaned toward pointer receivers.

Reply:
Yeah, we have a lack of consistency on that. I agree that we can lean towards pointer-receivers going forward.